src/btree.c

```diff
@@ -3021,6 +3041,15 @@ btree_pack(btree_pack_req *req)
    {
       iterator_get_curr(req->itor, &key, &data);
       btree_pack_loop(req, key, data, &at_end);
+
+      if (!at_end) {
```

Comment:
I don't think this should be guarded by `!at_end`.

Reply:
@rtjohnso -- Got it. As I change the code for `btree_pack_loop` to return an error, this guard is also removed.